Assets/Scripts/ConnectionManagement/ConnectionManager.cs

```diff
@@ -49,6 +55,21 @@ public class ConnectionPayload
     public class ConnectionManager : MonoBehaviour
     {
         ConnectionState m_CurrentState;
+        ConnectionState CurrentState
+        {
+            get => m_CurrentState;
+            set
+            {
+                if (m_CurrentState != null)
+                {
+                    m_CurrentState.StateChangeRequest -= OnStateChangeRequest;
```

Comment:
not sure I understand why this is happening, let's have some comments.
Why is this now a callback?

Reply:
The goal was to make it more robust and prevent classes that should not be modifying the state from doing it. Now only the current state it is subscribed to can request a state change.

Reply:
This seems like a good use for restricting access with "internal" and not add complexity to our API. (let your compiler do the work for you).